views/xTree/XTreeView.js

```diff
@@ -0,0 +1,1323 @@
+        Since recombination occurs most of the time, though not all of the time, when a mother passes along her X-Chromosome to her children,<br/>
+        the range that anyone can inherit from a single ancestor can be as low as 0%, or it could be higher than the average shown here.<br/><br/>
+        In this Probability Tree, each bar represents that probability of a piece coming from a direct ancestor, with the assumption that<br/>
+        at every point there IS a recombination, and that each Mother passes along to her children 50% of each of her X-Chromosomes.<br/><br/>
+        Though mathematically aesthetically pleasing, this EXACT result is highly unlikely.  <br/>
+        However, it should give an idea of which ancestors are more likely than others to have passed on some of their X-Chromosome.<br/>`;
+
+        XTreeView.detailsRandomSimulation = `<button onclick="XTreeView.changeMode(1);" style="padding:2px;">Randomize</button><br/>This Random Simulation generates one possible X-Chromosome Family Tree.<br/> This Family Tree demonstrates how the X-Chromosome can be passed along from ancestors to descendants.<br/><br/>
+        Since recombination occurs most of the time, though not all of the time, when a mother passes along her X-Chromosome to her children,<br/>
+        the range that anyone can inherit from a single ancestor will vary. She can pass on one or the other of her X-chromosomes, or a combination of both.<br/><br/>
+        In this Family Tree, there are red lines to indicate the recombination points, where the X-Chromosome being passed along switches from one to the other.<br/>
+        Note that if the only recombination indicators are at the beginning or end of the line, that indicates one is being passed along fully, and the other one ignored.<br/><br/>
+        Though visually pleasing, this EXACT result is highly unlikely.  <br/><br/>
+        However, it should give an idea of which ancestors are more likely than others to have passed on some of their X-Chromosome,<br/>
+        and make it obvious how random DNA inheritance really is.<br/><br/>
+        Comparing your X-chromosome DNA with other relatives' DNA is the only way to confirm which ancestors truly passed on theirs to you.<br/>`;
+
+        XTreeView.forMore = `<br/><hr/>For more details about the X-Chromosome inheritance pattern, check out the <a target=_blank href="https://youtu.be/ppO7tzxuik8?t=64">intro to this video</a>.<br/>
+        You can display your own X-Chromosome ancestors in the <a target=_blank href="#view=fanchart">Fan Chart</a> &rarr; open the Settings, choose Highlights, then Highlight cells by <I>X-chromosome inheritance</I>.<br/>
+        To investigate further and find other relatives that could share some of your ancestors X-Chromosomes, try the <a target=_blank href="https://apps.wikitree.com/apps/clarke11007/Xfriends.php">X-Friends app</a>.`;
```

Comment:
here is just an idea (when I see those large paragraphs of plain text) - there is posibility to store it in text file and load it "on the fly"

```js
await (await fetch("tree.js")).text()
```

(you can try it eg at wikitree.com in js console 😉)

Reply:
Thanks for all those suggestions Michal!  Some of the code was hanging around from another view that I "stole" this setup from ... so I can't claim it was all original - but - many of your suggestions WERE mine.  I appreciate you helping me make it more efficient.  The SVG using CSS ... I kind of knew that, but hadn't thought it through to the logical conclusion - makes a lot of sense.